### PR TITLE
LPS-204549 | Enable only supported field types for CSV export via UI

### PR DESCRIPTION
**Pull request by Hijky**
https://liferay.atlassian.net/browse/LPS-204549

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

File: modules/apps/batch-planner/batch-planner-test/src/testFunctional/tests/EnableOnlySupportedFieldTypesForCSVExportViaUI.testcase [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			ObjectDefinitionAPI.createObjectDefinitionWithMuliteCustomFields(
+				en_US_label = "test",
+				en_US_plural_label = "tests",
+				objectDefinitionName = "Test",
+				picklistId = ${picklistId},
+				requiredStringFieldName = "testText");
+		}
+
+		task ("And Given navigate to export in Data Migration Center") {
+			ImportExport.openImportExportAdmin();
```
Comment (yanan) [inline modules/apps/batch-planner/batch-planner-test/src/testFunctional/tests/EnableOnlySupportedFieldTypesForCSVExportViaUI.testcase:33]: We can try to move `ImportExport.openImportExportAdmin();` into gotoExport()

File: modules/apps/batch-planner/batch-planner-test/src/testFunctional/tests/EnableOnlySupportedFieldTypesForCSVExportViaUI.testcase [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		property portal.acceptance = "true";
+
+		task ("And Given selecting custom object entity type with CSV format") {
+			ImportExport.configureExport(
+				entityType = "C_Test (v1_0 - Liferay Object REST)",
+				exportFileFormat = "CSV");
+		}
+
+		task ("When I checked Attribute Code") {
+			Uncheck.uncheckNotVisible(
```
Comment (yanan) [inline modules/apps/batch-planner/batch-planner-test/src/testFunctional/tests/EnableOnlySupportedFieldTypesForCSVExportViaUI.testcase:85]: Is this uncheck necessary?

File: modules/apps/batch-planner/batch-planner-test/src/testFunctional/tests/EnableOnlySupportedFieldTypesForCSVExportViaUI.testcase [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+				locator1 = "Checkbox#SPECIFIC_CHECKBOX_IN_TABLE");
+
+			Check.checkNotVisible(
+				key_checkboxOption = "Attribute Code",
+				locator1 = "Checkbox#SPECIFIC_CHECKBOX_IN_TABLE");
+		}
+
+		task ("Then all supported fields are selected") {
+			for (var fieldNameList : list "testDateAndTime,testDate,testDecimal,testInteger,testLongInteger,testLongText,testRichText,testText") {
+				Check.checkNotVisible(
```
Comment (yanan) [inline modules/apps/batch-planner/batch-planner-test/src/testFunctional/tests/EnableOnlySupportedFieldTypesForCSVExportViaUI.testcase:96]: This `Check.checkNotVisible` is actually to check the field if it is not checked, we need to asset the field is actually in checked status not to check it

File: modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro [hunk 4]
```diff
@@ -838,6 +838,64 @@ definition {
 			objectDefinitionId = ${objectDefinitionId});
 	}
 
+	@summary = "Default summary"
+	macro createObjectDefinitionWithMuliteCustomFields(picklistId = null, objectDefinitionName = null, en_US_label = null, requiredStringFieldName = null, en_US_plural_label = null) {
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro:842]: typo

File: modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro [hunk 5]
```diff
@@ -838,6 +838,64 @@ definition {
...
+			requiredStringFieldName = ${requiredStringFieldName});
+
+		ObjectDefinitionAPI.postObjectDefinitionObjectField(
+			businessType = "Picklist",
+			dBType = "String",
+			listTypeDefinitionId = ${picklistId},
+			name = "testPicklist",
+			objectDefinitionName = ${objectDefinitionName});
+
+		ObjectFieldAPI.createObjectFieldByObjectDefinitionId(
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro:858]: We can try to use for loop as we talked about

**Review by yanan — COMMENTED** [review]
Thanks for the codes, please check the upper comments.

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Review by JoyceWang08211 — APPROVED** [review]
LGTM

**Review by magjed4289 — APPROVED** [review]

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**